packages/nameguard-sdk/src/index.ts

```diff
@@ -667,6 +678,10 @@ class NameGuard {
       );
     }
 
+    if (this.localProvider) {
+      return this.localProvider.securePrimaryName(address);
+    }
```

Comment:
The problem is that the `network` option will get ignored if a user sets the local provider. Not sure how to solve this.

Reply:
@Carbon225 Here's a recommendation: I suggest we create the following invariant:

If a `localProvider` is defined, the network of the `localProvider` must match the network of the NameGuard instance.

Suggest establishing this invariant in the constructor. Therefore it will be impossible that the case you described above could happen.

If this invariant is violated in the constructor the constructor should throw an `Error`.

Reply:
In the end I could not implement this. I could not find how to identify the network name from a viem client instance, especially if the user can pass a custom endpoint. I tried an approach where the user has to manually declare the name of the network in the constructor of the local provider but encountered another problem.

```ts
...
  securePrimaryName(address: string, network: Network): Promise<SecurePrimaryNameResult> {
    if (network !== this.network) {
      throw new Error("Network mismatch between LocalProvider and NameGuard SDK client. Please ensure the network is the same for both. NameGuard network: " + network + ", LocalProvider network: " + this.network + ".");
    }
    return securePrimaryNameImpl(address, this.publicClient);
  }
...
```

This always throws:
Error: Network mismatch between LocalProvider and NameGuard SDK client. Please ensure the network is the same for both. NameGuard network: **undefined**, LocalProvider network: mainnet.

It's used like this in nameguard sdk:

```ts
    const network_name = options?.network || this.network;

    if (this.localProvider) {
      return this.localProvider.securePrimaryName(address, network_name);
    }
```

I tried manually changing the `network_name` but it's always `undefined`.
It's probably some stupid mistake but I have no idea how to fix it.

Anyway, I'm not a fan of having to provide the network name manually when creating the local provider.
I think we should further investigate getting the network from the viem client.

There is this warning though:

```ts
/**
 * Options for the NameGuard LocalProvider constructor.
 */
interface LocalProviderOptions {
  /**
   * The viem client instance to use for primary name lookups.
   * The network of this client must match the network of the NameGuard client.
   */
  publicClient: PublicClient;
```

So it's made clear that the network should match and that is the responsibility of the user. For now we just cannot enforce it with an exception.

Reply:
@lightwalker-eth  Came up with this:

```typescript
export interface LocalProviderOptions {
  /**
   * The viem client instance to use for primary name lookups.
   * The network of this client must match the network declared in NameGuard's `createClient` options.
   * 
   * ```typescript
   * const nameguard = createClient({
   *   localProviders: {
   *     // The network of the publicClient must match the network declared here.
   *     "mainnet": createLocalProvider({ publicClient }),
   *     "sepolia": ...
   *   }
   * });
   * ```
   * 
   * It is the user's responsibility to ensure that the network of the viem client matches the declared network.
   */
  publicClient: PublicClient;
}
```

Now we can have multiple local providers for many networks.